### DISCUSSION TITLE
How does django-easy-tenants differ from django-tenants?

### DISCUSSION THREAD

**Discussion by thraizz** [CleitonDeLima/django-easy-tenants#109] [General]
Hi, we are currently prototyping different approaches for implementing a multi-tenancy backend and have played around with django-tenants, but just found this repository. How does it compare to django-tenants, what benefits does it bring? Thanks in advance :)

**Comment by CleitonDeLima** [comment]
> Olá, atualmente estamos prototipando diferentes abordagens para implementar um back-end multilocação e brincamos com django-tenants, mas acabamos de encontrar este repositório. Como ele se compara aos inquilinos do django, quais benefícios ele traz? Agradecemos antecipadamente :)

Hello,

`django-tenants` works with the postgresql database, each tenant would have its own schema (table structures for the data that will be isolated). I used [django-tenant-schemas](https://github.com/bernardopires/django-tenant-schemas/tree/master) for a long time (I think django-tenants is a continuation).

I created `django-easy-tenants` out of necessity in my saas application. Working with schemas became a problem almost as the customer base began to grow rapidly. As the application was in the growth phase, any migration took a long time to apply to all schemas.
One solution we found was to remove the schemas and leave all the data with normal tables (if the data is isolated per tenant it has a `tenant_id`). Instead of applying filters by tenant id everywhere, the library does this for you.

Another similar solution is [django-scopes](https://github.com/raphaelm/django-scopes) (my inspiration for creating this package), it works more with security by filtering data for each tenant.

`django-easy-tenants` is simpler to write the code because of [this](https://github.com/CleitonDeLima/django-easy-tenants/blob/main/easy_tenants/models.py#L14), it abstracts a lot of things right away of declaring querysets in class attributes (a common case used in django forms). I'm not an expert to say whether this is correct or not, but it solved my problem.
